### ISSUE TITLE
Type List for OpenApiPayloadAttribute not mapped correctly

### ISSUE THREAD

**Issue by ChristianWeyer** [aliencube/AzureFunctions.Extensions#34]
Hi,

I have a Function like this, returning a `List<Product>`:
```
[FunctionName("ListProducts")]
[OpenApiOperation("products", "Serverless Microservices")]
**[OpenApiResponseBody(HttpStatusCode.OK, "application/json", typeof(List<Product>))]**
public async Task<IActionResult> ListProducts...
```

The Product type looks like this:
```
public class Product
{
        public Guid Id { get; set; }
        public string Name { get; set; }
}
```

The JSON rendered for this response type is this:
```
{
  **"List`1"**: {
    "type": "array",
    "items": {
      "type": "object",
      "properties": {
        "Id": {
          "format": "uuid",
          "type": "string"
        },
        "Name": {
          "type": "string"
        }
      }
    }
  }
}
```
... which is not really what I expected ;-)
Shouldn't there be a `Product` type and then a list/array of `Product` in the JSON?

Thanks!

**Comment by justinyoo** [comment]
@ChristianWeyer Thanks for raising this issue! I'll have a look and get back to you.

**Comment by hf-kklein** [comment]
The problem also applies for list types in the `OpenApiRequestBody`.
@ChristianWeyer I'd suggest to change the title of this issue to 'Type List<T> for OpenApiPayloadAttribute not mapped correctly'.

**Comment by justinyoo** [comment]
I’m not too sure if this boils be with `Product` or not. What if the type is `List<int>`?

I’ve excluded `JObject` and `Dictionary<>` from being a root level payload type, at #29. I think so does the `List<>`, too.


Kind regards,

Justin Yoo | Microsoft MVP | aliencube.org | devkimchi.com

**Comment by hf-kklein** [comment]
> I’ve excluded `JObject` and `Dictionary<>` from being a root level payload type, at #29. I think so does the `List<>`, too.

By saying so, do you consider this issue as a missing feature/bug or being out of scope?

IMHO returning list types is a valid use case and the generated OpenApi.json should render somehow like 
```json
"content": {
    "application/json": {
      "schema": {
        "type": "array",
        "items": {
          "$ref": "#/components/schemas/Product"
        }
      }
    }
```
and include a schema derived from `class Product`.

**Comment by glennular** [comment]
The support for List<> is part of #37.

**Comment by justinyoo** [comment]
@ChristianWeyer This has been fixed and released to [1.5.2](https://www.nuget.org/packages/Aliencube.AzureFunctions.Extensions.OpenApi/1.5.2)

**Comment by ChristianWeyer** [comment]
Sorry, this does not yet seem to work?

I am using 1.5.2 now.
The Swagger JSON now looks like this:
```
{
  "swagger": "2.0",
  "info": { },
  "host": "localhost:7071",
  "schemes": [
    "http"
  ],
  "paths": {
    "/api/products": {
      "get": {
        "tags": [
          "Serverless Microservices"
        ],
        "operationId": "products",
        "produces": [
          "application/json"
        ],
        "responses": {
          "200": {
            "description": "Payload of Array of Product",
            "schema": {
              "type": "array",
              "items": {
                "$ref": "#/definitions/Product"
              }
            }
          }
        }
      }
    },
    "/api/products/{id}": {
      "get": {
        "tags": [
          "Serverless Microservices"
        ],
        "operationId": "productbyid",
        "produces": [
          "application/json"
        ],
        "responses": {
          "200": {
            "description": "Payload of ProductDetails",
            "schema": {
              "$ref": "#/definitions/ProductDetails"
            }
          }
        }
      }
    }
  },
  "definitions": {
    "ProductDetails": {
      "type": "object",
      "properties": {
        "Id": {
          "format": "uuid",
          "type": "string"
        },
        "Name": {
          "type": "string"
        },
        "Description": {
          "type": "string"
        },
        "OnStock": {
          "type": "boolean"
        }
      }
    }
  },
  "securityDefinitions": {
    "authKey": {
      "type": "apiKey",
      "name": "x-functions-key",
      "in": "header"
    }
  }
}
```

As you can see, there is no definition for Product in there.
This seems to be missing - and thus the UI breaks as well:

![image](https://user-images.githubusercontent.com/888718/58817414-f5f99100-862b-11e9-82e8-d5b5fecbc0b7.png)

Any idea what could be wrong here? @justinyoo @glennular 
Thanks!
